frontend/src/app/(navfooter)/settings/ai/AISettings.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import {
+    AIProvider,
+    AIModel,
+    useAIProvider,
+    useAIModel,
+    useAIAPIKey,
+} from "@/lib/settings";
+import RadioList from "@/app/(navfooter)/settings/RadioList";
+import Section from "@/app/(navfooter)/settings/Section";
+import TextField from "@/app/(navfooter)/settings/TextField";
+
+const providerOptions = {
+    [AIProvider.OPEN_AI]: {
+        label: <div>OpenAI</div>,
+    },
+    [AIProvider.DEEP_SEEK]: {
+        label: <div>DeepSeek</div>,
+    },
+};
+
+const openAIModels = {
+    [AIModel.O1_PREVIEW]: {
+        label: <div>o1-preview (recommended)</div>,
+    },
+    [AIModel.GPT_3_5_TURBO]: {
+        label: <div>GPT 3.5 Turbo</div>,
+    },
+};
+const openAIRecommendedModel = AIModel.O1_PREVIEW;
+
+const deepSeekAIModels = {
+    [AIModel.DEEPSEEK_REASONER]: {
+        label: <div>deepseek-reasoner (recommended)</div>,
+    },
+    [AIModel.DEEPSEEK_CHAT]: {
+        label: <div>deepseek-chat</div>,
+    },
+};
+const deepSeekRecommendedModel = AIModel.DEEPSEEK_REASONER;
+
+export default function AISettings() {
+    const [aiProvider, setAIProvider] = useAIProvider();
+    const [aiModel, setAIModel] = useAIModel();
+    const [aiAPIKey, setAIAPIKey] = useAIAPIKey();
+
+    const modelOptions =
+        aiProvider === AIProvider.OPEN_AI ? openAIModels : deepSeekAIModels;
```

Comment:
Better to use an object to look this up, since we are likely to add providers in future